src/Dapr.Client/EnumExtensions.cs

```diff
@@ -0,0 +1,37 @@
+﻿// ------------------------------------------------------------------------
+// Copyright 2023 The Dapr Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+using System;
+using System.Runtime.Serialization;
+
+namespace Dapr.Client
+{
+    internal static class EnumExtensions
+    {
+        /// <summary>
+        /// Reads the value of an enum out of the attached <see cref="EnumMemberAttribute"/> attribute.
+        /// </summary>
+        /// <typeparam name="T">The enum.</typeparam>
+        /// <param name="value">The value of the enum to pull the value for.</param>
+        /// <returns></returns>
+        public static string GetValueFromEnumMember<T>(this T value) where T : Enum
+        {
+            var memberInfo = typeof(T).GetMember(value.ToString());
```

Comment:
Consider constraining the types of members returned to be *only* the static/public/declared fields of that enumerated type (e.g. not base types, such as `Object`).

Reply:
Done